docs/rules/no-classic-classes.md

```diff
@@ -0,0 +1,44 @@
+# Disallow "classic" classes in favor of native JS classes (no-classic-classes)
+
+Ember now allows you to use native JS classes to extend the built-in classes provided by Ember. This pattern is preferred in favor of using the "classic" style of classes that Ember has provided since before JS classes were available to use.
+
+## Rule Details
+
+This rule aims to ensure that you do not use a "classic" Ember class where a native class could be used instead. The one instance where `.extend` should still be used is for including a Mixin into your class, which does not have a native JS class alternative available.
+
+Examples of **incorrect** code for this rule:
+
+```javascript
+// Extending an Ember class using the "classic" class pattern is not OK
+import Component from '@ember/component';
+
+export default Component.extend({});
+```
+
+Examples of **correct** code for this rule:
+
+```javascript
+// Extending using a native JS class is OK
+import Component from '@ember/component';
+
+export default class MyComponent extends Component {};
+```
+
+```javascript
+// Including a Mixin is OK
+import Component from '@ember/component';
+import Evented from '@ember/object/evented';
+
+export default class MyComponent extends Component.extend(Evented) {};
+```
+
+## When Not To Use It
+
+* If you are not ready to transition completely to native JS classes, you should not enable this rule
+
+## Further Reading
```

Comment:
Can you add links to the official documentation (in addition to the blog posts)?

Reply:
Which docs do you want me to link to?

The 3.13 Object Model docs still reference classic classes

Reply:
That's fine. I thought there would be official documentation on new classes somewhere.

Reply:
Not yet, that I know of -- I believe the Octane-ified docs are WIP, to be released with Ember 3.14.